src/Confluent.Kafka/Consumer.cs

```diff
@@ -734,7 +732,11 @@ public void Stop()
         {
             if (consumerCts == null)
             {
-                throw new Exception("Consumer background poll loop not started - cannot stop.");
+                // Consumer background poll loop not started - cannot stop.
+                // While this might be considered an error there is no point
+                // in throwing an exception here since there are no side-effects
+                // of not being able to stop a non-Started instance.
+                return;
```

Comment:
The problem stems from (auto) disposing a consumer without calling Stop() first: it will leave threads behind and eventually assert on some refcount issue.
I also think we should be forgiving in APIs like these, otherwise app developers will need to keep track state (did I start or not?) that we already have.